packages/shorebird_cli/lib/src/commands/releases/get_apk_command.dart

```diff
@@ -0,0 +1,186 @@
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_validator.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template get_apk_command}
+/// Generates an APK for the release with the specified version.
+/// {@endtemplate}
+class GetApkCommand extends ShorebirdCommand {
+  /// {@macro get_apk_command}
+  GetApkCommand() {
+    argParser
+      ..addOption(
+        CommonArguments.releaseVersionArg.name,
+        help: 'The release version to generate an apk for',
+      )
+      ..addOption(
+        CommonArguments.flavorArg.name,
+        help: 'The build flavor to generate an apk for',
+      )
+      ..addOption(
+        'out',
+        abbr: 'o',
+        help: 'The output directory for the generated apks',
+      );
+  }
+
+  @override
+  String get name => 'get-apk';
+
+  @override
+  String get description =>
+      'Generates a universal apk for the specified release version';
+
+  /// The shorebird app ID for the current project.
+  String get appId => shorebirdEnv.getShorebirdYaml()!.getAppId(flavor: flavor);
+
+  /// The build flavor, if provided.
+  late String? flavor = results.findOption(
+    CommonArguments.flavorArg.name,
+    argParser: argParser,
+  );
+
+  /// The output directory path for the generated apks. Defaults to the
+  /// project's build directory if not provided.
+  late String? outDirectoryPath = results.findOption(
+    'out',
+    argParser: argParser,
+  );
+
+  @override
+  Future<int> run() async {
+    try {
+      await shorebirdValidator.validatePreconditions(
+        checkUserIsAuthenticated: true,
+        checkShorebirdInitialized: true,
+      );
+    } on PreconditionFailedException catch (error) {
+      return error.exitCode.code;
+    }
+
+    final Release release;
+    if (results.wasParsed(CommonArguments.releaseVersionArg.name)) {
+      final releaseVersion =
+          results[CommonArguments.releaseVersionArg.name] as String;
+      release = await codePushClientWrapper.getRelease(
+        appId: appId,
+        releaseVersion: releaseVersion,
+      );
+    } else {
+      release = await _promptForRelease();
+    }
+
+    final releaseArtifact = await codePushClientWrapper.getReleaseArtifact(
+      appId: appId,
+      releaseId: release.id,
+      arch: 'aab',
+      platform: ReleasePlatform.android,
+    );
+
+    final aabFile = await _downloadAab(releaseArtifact: releaseArtifact);
+    final apksFile = File(
+      p.join(
+        Directory.systemTemp.createTempSync().path,
+        '${appId}_${release.version}.apks',
+      ),
+    );
+
+    final buildApksProgress = logger.progress(
+      'Building apks for release ${release.version} (app: $appId)',
+    );
+    try {
+      await bundletool.buildApks(bundle: aabFile.path, output: apksFile.path);
```

Comment:
`shorebird preview` has almost identical logic so it would be nice to refactor to share more code rather than having this implementation live in two separate commands

Reply:
Happy to, although the code shared between here and preview is split across several different places to accommodate both iOS and Android, and has a preview-specific cache that I was a bit hesitant to interact with here.

Reply:
This specific bit of code has also diverged, as preview always generates a universal APK and this command offers a way to disable that, so I'm less certain how much can be shared